openwisp_monitoring/device/migrations/0009_update_device_status_for_disabled_critical_checks.py

```diff
@@ -0,0 +1,27 @@
+from django.conf import settings
+from django.db import migrations
+from swapper import load_model
+
+Check = load_model('check', 'Check')
+Device = load_model('config', 'Device')
+DeviceMonitoring = load_model('device_monitoring', 'DeviceMonitoring')
+
+
+def update_device_status(apps, schema_editor):
+    critical_metrics = settings.OPENWISP_MONITORING_CRITICAL_DEVICE_METRICS
+    for check in Check.objects.filter(is_active=False):
```

Comment:
1. let's use the `iterator()` method to avoid this line to potentially crash big databases
2. let's filter checks properly, eg: `Check.objects.filter(is_active=False, metric__key__in=[critical_metrics_key])`, you'll need to prepare `critical_metrics_key` beforehand as `OPENWISP_MONITORING_CRITICAL_DEVICE_METRICS` is a list of dictionaries.